jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -130,13 +130,15 @@ fun loginSso(project: Project?, startUrl: String, scopes: List<String> = ALL_SON
         }
 
         // There is an existing connection we can use
-        if (connection is BearerSsoConnection && !scopes.all { it in connection.scopes }) {
+        if (connection is BearerSsoConnection && !requestedScopes.all { it in connection.scopes }) {
+            allScopes.addAll(connection.scopes)
+
             logger.info {
-                "Forcing reauth on ${connection.id} since requested scopes ($scopes) are not a complete subset of current scopes (${connection.scopes})"
+                "Forcing reauth on ${connection.id} since requested scopes ($requestedScopes) " +
+                    "are not a complete subset of current scopes (${connection.scopes})"
```

Comment:
nit: formatting

Reply:
The identation is required otherwise detekt complains

Reply:
i dont think it'll complain if you make it a multi-line string.
you can also can just suppress the length error
https://pinterest.github.io/ktlint/faq/#how-do-i-suppress-errors-for-a-lineblockfile

```kt
/* ktlint-disable max-line-length */
...
/* ktlint-enable max-line-length */
```

```kt
looooooooooooooooooooooong // ktlint-disable max-line-length
```